### DISCUSSION TITLE
Getting invalid claim: missing sub claim when trying to linkIdentity of my anonymous logged in user via google oauth

### DISCUSSION THREAD

**Discussion by AbdullahMuhammad** [supabase/supabase#23201] [Questions]
I log in my user anomymously and save the access token in the cookie.
 `const { data: signInData, error: signInError } = await supabase.auth.signInAnonymously();`
Which is working pretty fine. 

I want to convert my anonymous user to permanent user.
https://supabase.com/docs/guides/auth/auth-anonymous

My code snipper is as follows.

       ```
const { data, error } = await supabase.auth.linkIdentity({
            provider: provider as Provider,
            options: { redirectTo: authRedirect, },
        });

        if (error) {
            return new Response(error.message, { status: 500 });
        }

        return redirect(data.url);

```
Instead of redirecting me to google which is my provider it redirects me to a page with the following error.

> invalid claim: missing sub claim

Note if I do supabase.auth.signInWithOAuth with the same parameters its works fine. 
But i need to make my anomymous user permanent.

Any help is highly appreciated.

**Comment by adao** [comment]
did you ever figure it out ?

I have an anonymouse user in my session, then I do `supabase.auth.linkIdentity({provider: 'google'})` and it goes through the flow and redirects me to my home page with 

https://dev.browserbuddies.co?state=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MjM3MzgzMzYsInNpdGVfdXJsIjoiaHR0cHM6Ly9kZXYuYnJvd3NlcmJ1ZGRpZXMuY28iLCJpZCI6IjAwMDAwMDAwLTAwMDAtMDAwMC0wMDAwLTAwMDAwMDAwMDAwMCIsImZ1bmN0aW9uX2hvb2tzIjpudWxsLCJwcm92aWRlciI6Imdvb2dsZSIsInJlZmVycmVyIjoiaHR0cHM6Ly9kZXYuYnJvd3NlcmJ1ZGRpZXMuY28vIiwiZmxvd19zdGF0ZV9pZCI6IiIsImxpbmtpbmdfdGFyZ2V0X2lkIjoiMTk2YzllMDktMTVhNC00MDVkLTkxYjYtNjg3MDFhMGZiOGU3In0.3_OeG5CAkAaVKnCDsuZsnQXghiR68KP1w4fCS5zRrMo&code=4%2F0AcvDMrCOIzLMaqyYPPeRNPXrV3JJBXVrJ0pJd0SpEaXoPSfwbIY0oqmeb3dPyv4zaYNm5A&scope=email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+openid&authuser=1&prompt=none

but i don't know what to do from here. my session is still anonymous

**Comment by AbdullahMuhammad** [comment]
Yes I figured it out. I would need to recall a few things and also go
through my code because it has been some time.
I will surely get back to you by tomorrow.

On Thu, Aug 15, 2024 at 9:15 PM Anthony ***@***.***> wrote:

> did you ever figure it out ?
>
> I have an anonymouse user in my session, then I do supabase.auth.linkIdentity({provider:
> 'google'}) and it goes through the flow and redirects me to my home page
> with
>
>
> https://dev.browserbuddies.co?state=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MjM3MzgzMzYsInNpdGVfdXJsIjoiaHR0cHM6Ly9kZXYuYnJvd3NlcmJ1ZGRpZXMuY28iLCJpZCI6IjAwMDAwMDAwLTAwMDAtMDAwMC0wMDAwLTAwMDAwMDAwMDAwMCIsImZ1bmN0aW9uX2hvb2tzIjpudWxsLCJwcm92aWRlciI6Imdvb2dsZSIsInJlZmVycmVyIjoiaHR0cHM6Ly9kZXYuYnJvd3NlcmJ1ZGRpZXMuY28vIiwiZmxvd19zdGF0ZV9pZCI6IiIsImxpbmtpbmdfdGFyZ2V0X2lkIjoiMTk2YzllMDktMTVhNC00MDVkLTkxYjYtNjg3MDFhMGZiOGU3In0.3_OeG5CAkAaVKnCDsuZsnQXghiR68KP1w4fCS5zRrMo&code=4%2F0AcvDMrCOIzLMaqyYPPeRNPXrV3JJBXVrJ0pJd0SpEaXoPSfwbIY0oqmeb3dPyv4zaYNm5A&scope=email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+openid&authuser=1&prompt=none
>
> but i don't know what to do from here. my session is still anonymous
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/supabase/discussions/23201#discussioncomment-10349850>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAH3HYZRTNRTQSOSIHMICDDZRTH2XAVCNFSM6AAAAABGVQ7GBKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZUHE4DKMA>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***
> com>
>


-- 
ALLAH GREAT

**Reply by adao** [reply]
Thanks! I finally figured it out. I needed to redirect back to my Supabase server at a specific path `/auth/v1/callback`. And, for self-hosted version, I needed to set SITE_URL environment variable so Supabase Auth server knows the client app URL to redirect back to after linking the new identity.
